### ISSUE TITLE
Release build troubles on iOS

### ISSUE THREAD

**Issue by ray007** [nativescript-community/ui-material-components#242]
### Which platform(s) does your issue occur on?
- iOS, iPad device

### Please, provide the following version numbers that your issue occurs with:

- CLI: 7.2.0
- Cross-platform modules: 7.2.1
- iOS Runtime: 7.2.0
- XCode Version: 12.4
- Plugin(s): ```
"dependencies": {
    "@bradmartin/nativescript-urlhandler": "^2.0.1",
    "@nativescript-community/ble": "^3.0.14",
    "@nativescript-community/ui-material-bottomnavigationbar": "^5.2.5",
    "@nativescript-community/ui-material-bottomsheet": "^5.2.5",
    "@nativescript-community/ui-material-button": "^5.2.5",
    "@nativescript-community/ui-material-progress": "^5.2.5",
    "@nativescript-community/ui-share-file": "^1.2.1",
    "@nativescript/appversion": "^2.0.0",
    "@nativescript/core": "^7.2.1",
    "@nativescript/theme": "^3.0.1",
    "@nota/nativescript-webview-ext": "^7.0.3",
    "@sergeymell/nativescript-svg": "^1.0.5",
    "http-auth-client": "^0.4.1",
    "js-base64": "^3.6.0",
    "nativescript-downloader": "^2.1.5",
    "nativescript-inappbrowser": "^3.0.2",
    "nativescript-permissions": "^1.3.11",
    "nativescript-ui-listview": "^9.1.0",
    "nativescript-ui-sidedrawer": "^9.0.3",
    "nativescript-vue": "^2.8.3",
    "url-parse": "^1.4.7",
    "url-search-params-polyfill": "^8.1.0"
  },```

### Please, tell us how to recreate the issue in as much detail as possible. 
Testing with `ns debug ios` and `ns run ios` everything worked fine, but trying `ns run ios --release` has the app crashing.
Which I first noticed when trying to upload my build to appstore, see https://github.com/NativeScript/nativescript-cli/issues/5499

After doing a clean checkout and trying again `ns run ios --release`, I noticed the build system now tried to install different Pods compared to the debug build.
Instead of
```
Installing MDFInternationalization (2.0.0)
Installing MDFTestAccessibility (2.0.0)
Installing MaterialComponents (122.0.0)
```
I now see
```
Installing MDFInternationalization (2.0.0)
Installing MDFTestAccessibility (2.0.0)
Installing MaterialComponents (94.5.0)
Installing MotionAnimator (2.8.1)
Installing MotionInterchange (1.6.0)
```
And the application crashes with:
```
****** Fatal JavaScript exception - application has been terminated *****
NativeScript encountered fatal error: Uncaught TypeError: this.nativeViewProtected.startAnimating is not a function
at [indeterminate:setNative] (file:///....)
...
```

### Source of the problem
Not using the `indeterminate` mode of the linear progress widget makes `ns run ios --release` work again.
Activating `indeterminate` mode with a property also does not work.

**Comment by ray007** [comment]
Note: doing `ns build ios --release` instead of `ns run ios --release` installs the same Pods as debugging.
I'm now officially very confused.

**Comment by farfromrefug** [comment]
@ray007 something is wrong with your setup. With the latest material components you should have `MaterialComponents ` at the latest versions. 95.0 was a version forced by N. I have a hook removing that forced version in materila components 5.x
Check that the material component core hook is correctly running. 5.x is not compatible with `MaterialComponents `  95.0

**Comment by ray007** [comment]
Yes, I'm aware, but other than keeping the Mac up-to-date, I haven't done anything which would explain the problems popping up.

It's also interesting to note that `ns run ios --release` and `ns deploy ios --release` are dragging in the old version, while `ns build ios --release` does not.

The core hook scripts is `@nativescript-community/ui-material-core/scripts/before-prepare.js`? Or is there something else?
I know I see the warning about `Using material components 5.x ...`, is there something else I should look for?

And even after a build with the newer `MaterialComponents` I seems to have a problem with the linker complaining about missing framework (due to missing directories in `.../Library/Developer/Xcode/DerivedData/...`, see also [linked issue](https://github.com/NativeScript/nativescript-cli/issues/5499)) which is only visible in XCode. Somehow the build-parameters for the Pods seem to be wrong, any idea how to find that problem?

Many thanks again for taking your time to help

**Comment by farfromrefug** [comment]
@ray007 your problem seems to be with building not this component. I will leave this open for a while though

**Comment by ray007** [comment]
Yes, the problem seems to be with building, but only with components using Pods that need to be build.
Do you have a suggestion for me where I'd better ask for help with that than here?

But the old version for the `ns run ios --release` seems to be a specific problem for material-ui, no?
